react/features/recent-list/functions.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+
+import { RECENT_URL_STORAGE } from './constants';
+import { parseURIString } from '../base/util';
+import moment from 'moment';
+
+/**
+* Retreives the recent room list and generates all the data needed
+* to be displayed.
+*
+* @returns {Promise} The {@code Promise} to be resolved when the list
+* is available.
+*/
+export function getRecentRooms(): Promise<Array<Object>> {
+    return new Promise(resolve => {
+        window.localStorage._getItemAsync(RECENT_URL_STORAGE)
+        .then(recentUrls => {
+
+            if (recentUrls) {
+                const recentUrlsObj = JSON.parse(recentUrls);
+                const recentRoomDS = [];
+
+                for (const entry of recentUrlsObj) {
+                    const location = parseURIString(entry.conference);
+
+                    if (location && location.room && location.hostname) {
+                        recentRoomDS.push({
+                            baseURL: `${location.protocol}//${location.host}`,
+                            conference: entry.conference,
+                            dateTimeStamp: entry.date,
+                            conferenceLength: entry.conferenceLength,
+                            dateString: _getDateString(entry.date),
+                            conferenceLengthString: _getLengthString(
```

Comment:
I think "duration" instead of length is a better term here

Reply:
Changed